chore(ci): move changelog fragment validation to vdev/Rust

pront opened the pull request:
## Summary

Ports `scripts/check_changelog_fragments.sh` to Rust as `vdev check changelog-fragments`, following the same pattern we used for the VRL checks. The Changelog CI workflow now installs vdev via `scripts/environment/prepare.sh --modules=vdev` (which uses `cargo binstall` when available) and invokes `vdev check changelog-fragments`. Locally, the check is exposed as `make check-changelog-fragments`.

Behavior is preserved:

- Same filename schema: `<unique_name>.<fragment_type>.md`
- Same fragment types: `breaking | security | feature | enhancement | fix`
- Same mandatory `authors:` trailer with `@`-prefix and comma-delimited rejection
- Same `--merge-base` default (`origin/master`) and 1000-fragment cap

Unblocks follow-up work to formalize the `breaking` fragment schema and auto-generate the release upgrade guide from those fragments.

## Vector configuration

N/A — CI/tooling change.

## How did you test this PR?

- `cargo build -p vdev` / `cargo clippy --all-targets -- -D warnings` clean
- `vdev check changelog-fragments` locally against master: correctly errors with no fragments
- Happy path: added a valid fragment, checker exits 0
- Sad paths (each exits non-zero with a targeted error): invalid fragment type, wrong extension, missing `authors:` line, `@` prefix, comma-delimited authors
- `make check-changelog-fragments` end to end

## Change Type

- [ ] Bug fix
- [ ] New feature
- [ ] Dependencies
- [x] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?

- [ ] Yes
- [x] No

## Does this PR include user facing changes?

- [ ] Yes. Please add a changelog fragment based on our [guidelines](https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md).
- [x] No. A maintainer will apply the `no-changelog` label to this PR.

thomasqueirozb commented on .github/workflows/changelog.yaml:
```diff
@@ -49,6 +49,7 @@ jobs:
           sparse-checkout: |
             scripts/check_changelog_fragments.sh
```
```suggestion
```

ghost commented:
<b>✅&nbsp;Tests</b>

<details><summary><b>🎉 All green!</b></summary>

❄️ No new **flaky tests** detected  
🧪 All **tests** passed
</details>

<sub>
This comment will be updated automatically if new data arrives. <br />
🔗 Commit SHA: e772661 | <a href="https://docs.datadoghq.com/integrations/guide/source-code-integration/#pr-comments">Docs</a> | Was this helpful? Give us <a href="https://forms.gle/ys6MDmdQEeC5pkzWA">feedback</a>!
</sub>

pront commented on vdev/src/commands/release/prepare.rs:
```diff
@@ -389,10 +390,6 @@ impl Prepare {
 
 // FREE FUNCTIONS AFTER THIS LINE
```
Aren't they? They are stateless and out of an `impl` block.

pront replied:
I also like to put such functions at the bottom of the file and start the file with the public interface. But we digress.

thomasqueirozb commented on .github/workflows/changelog.yaml:
```diff
@@ -58,7 +62,13 @@ jobs:
-          sparse-checkout: changelog.d/
+          sparse-checkout: |
+            changelog.d/
+            scripts/environment/prepare.sh # FIXME DO NOT MERGE
```
Reminder: this change needs to be reverted before merging

pront commented:
Let's wait until vdev is bin-installable.

pront committed:
chore(ci): port changelog fragment validation to vdev

Replaces scripts/check_changelog_fragments.sh with a Rust equivalent
under `vdev check changelog-fragments`. Wired into the Changelog CI
workflow via scripts/environment/prepare.sh (binstall path) and
exposed locally as `make check-changelog-fragments`.

Behavior is preserved: same filename schema, same fragment types
(breaking, security, feature, enhancement, fix), same mandatory
`authors:` trailer with the same @-prefix and comma checks.

thomasqueirozb commented on .github/workflows/changelog.yaml:
```diff
@@ -44,14 +44,16 @@ jobs:
 
+      - name: Install vdev
+        if: env.SHOULD_RUN == 'true'
+        run: bash scripts/environment/prepare.sh --modules=vdev
```
Use the setup action instead to pull vdev from the github cache

pront committed:
chore(ci): use setup action to install vdev in changelog workflow

pront committed:
chore(vdev): bump version to 0.3.8

pront committed:
chore(vdev): tighten changelog fragment checks

Reject fragments in changelog.d/ subdirectories and reject trailing
blank lines after the authors: trailer. Both cases would silently
break the release cue generator, which reads only direct children of
changelog.d and expects authors: on the raw last line.

pront commented on .github/workflows/changelog.yaml:
```diff
@@ -44,14 +44,16 @@ jobs:
-        run: |
-          # Compare against origin/master
-          ./scripts/check_changelog_fragments.sh
+        run: vdev check changelog-fragments
```
Fixed in vectordotdev/vector@b499685fc3.

pront commented on vdev/src/commands/check/changelog_fragments.rs:
```diff
@@ -0,0 +1,138 @@
+    let last_line = content
+        .lines()
+        .rev()
+        .find(|l| !l.trim().is_empty())
```
Fixed in vectordotdev/vector@cfa1a9be30.

pront commented on vdev/src/commands/check/changelog_fragments.rs:
```diff
@@ -0,0 +1,138 @@
+        }
+
+        for path in &fragments {
+            let Some(name) = path.file_name().and_then(|s| s.to_str()) else {
```
Fixed in vectordotdev/vector@cfa1a9be30.

pront commented on vdev/src/commands/check/changelog_fragments.rs:
```diff
@@ -0,0 +1,143 @@
+            let Some(name) = path.file_name().and_then(|s| s.to_str()) else {
+                bail!("Unexpected fragment path: {}", path.display());
+            };
+            if name == "README.md" {
+                continue;
```
We can live with this.
